File: 220/ruby/observer/stockchecker.rb
class StockChecker
  def initialize(level)
    @level = level
    @observers = []
  end

  def add_observer(observer)
    @observers << observer
  end
  
  def buy_item
    @level -= 1
    notify_observers
  end
  
  def notify_observers
    @observers.each { |obs| obs.notify(@level) }
  end
end

class StockObserver
  def notify(level)
    @level = level
  end

  def latest_notification
    @level
  end
end

File: 220/js/observer/stockchecker.js
StockChecker = function StockChecker(level) {
	var observers = [];
	this.addObserver = function (observer) {
		observers.push(observer);
	};
	this.buyItem = function () {
		level -= 1;
		observers.forEach(function (obs) {obs.notify(level); });
	};
};

StockObserver = function StockObserver() {
	var latest = 0;
	this.notify = function (level) {
		latest = level;
	};
	this.latestNotification = function () {
		return latest;
	};
};